### ISSUE TITLE
Forbidden

### ISSUE THREAD

**Issue by hg-smart** [OraOpenSource/apex-frontend-boost#94]
Hello all,

just wanted to use a Pre-Built VM from Oracle (Oracle DB Developer VM 2016-04-15) Oracle 64 Bit.

This VM uses Port Forwarding so you can run APEX direct from your localhost using Port 8080 and NAT. Ideal for Demoing ...

But when you run apex-frontend-boost you get an error (http 403 - Forbidden), is this a problem inside the VM

Best Hansjörg

{
    "hgdemo": {
        "appURL": "http://192.168.178.56:8080/ords/f?p=hgdemo"
....

 -----------------------------------------------------
       Local: http://localhost:3000/ords/f?p=hgdemo
    External: http://192.168.56.1:3000/ords/f?p=hgdemo
 -----------------------------------------------------
          UI: http://localhost:3001
 UI External: http://192.168.56.1:3001
 -----------------------------------------------------

HTTP 403 : Forbidden
The request cannot be processed because this resource does not support Cross Origin Sharing requests, or the request Origin is not authorized to access this resource. If ords is being reverse proxied ensure the front end server is propagating the host name, for mod_proxy ensure ProxyPreserveHost is set to On

**Comment by vincentmorneau** [comment]
Hi @hg-smart ,

We have encountered the same issue on one of our projects as well. I will investigate and get back soon. 

Thanks.

**Comment by vincentmorneau** [comment]
Hi @hg-smart , can you try something for me?

Assuming you are using v2.1.0, on gulpfile.js, after line 122, please add `res.setHeader('Access-Control-Allow-Origin', '*');` and try running your app again?

It appears like you app is requesting CORS to be enabled. I've had cases where CORS needed to be disabled (https://github.com/OraOpenSource/apex-frontend-boost/issues/70). I need to investigate further.

@tschf has encountered the same issue at a time we were enforcing `res.setHeader('Access-Control-Allow-Origin', '*');`. Any thoughts are welcome.

**Comment by hg-smart** [comment]
sorry for my late response, added the line ...

```
    apexMiddleware = function (req, res, next) {
        res.setHeader('Set-Cookie', ['oos-apex-frontend-boost-app-images=//' + req.headers.host + '/']);
        res.setHeader('Access-Control-Allow-Origin', '*');
        next();
    };
```

same error... forbidden
403 Forbidden
The request cannot be processed because this resource does not support Cross Origin Sharing requests, or the request Origin is not authorized to access this resource. If ords is being reverse proxied ensure the front end server is propagating the host name, for mod_proxy ensure ProxyPreserveHost is set to On

**Comment by vincentmorneau** [comment]
Thanks. I'll leave this here while investigating some more: 
- https://community.oracle.com/thread/3906326?start=0&tstart=0
- https://community.oracle.com/message/13659147#13659147

**Comment by vincentmorneau** [comment]
@hg-smart What ORDS version is your VM using?

**Comment by hg-smart** [comment]
`Oracle REST Data Services 3.0.4.60.12.48`

**Comment by vincentmorneau** [comment]
@hg-smart , do you think it would be possible for you to send me that VM in private so I can reproduce the issue myself? I need to create a test case somehow.

**Comment by hg-smart** [comment]
yes of course, do you have dropbox or google drive ?

**Comment by vincentmorneau** [comment]
Dropbox :) Please send an invite to vincent.morneau@gmail.com

**Comment by hg-smart** [comment]
this evening....

**Comment by vincentmorneau** [comment]
What I found so far:
- Upgrading to ORDS 3.0.5 didn't help. Error still occurs.
- Adding the static files to #APP_IMAGES#, and running it through port 8080 works fine, so it's not an isolated ORDS problem.

The error clearly points to an incompatibility between localhost:8080 and localhost:3000
> 403 Forbidden
> The request cannot be processed because this resource does not support Cross Origin Sharing requests, or the request Origin is not authorized to access this resource. If ords is being reverse proxied ensure the front end server is propagating the host name, for mod_proxy ensure ProxyPreserveHost is set to On

Going to investigate more.

**Comment by vincentmorneau** [comment]
@hg-smart I finally got around the problem.

- So your VM was setup with ORDS `3.0.4`. 
- We upgraded it to ORDS `3.0.5`, but no luck.
- Then we downgraded it to ORDS `3.0.2` (before people started having problems on the forum links above)
- And `3.0.2` worked. 

As a comparison, https://apex.oracle.com/ itself uses ORDS `2.0.x`, so I guess Oracle themselves don't think ORDS `3.0.x` is stable enough.

I will update the documentation to indicate ORDS `3.0.3`, `3.0.4` and `3.0.5` are not supported. Hopefully they release a version that re-enables Cross Origin Resource Sharing.

If that's okay with you @hg-smart, I will stop investigating.
